packages/flutter/lib/src/foundation/timeline.dart

```diff
@@ -0,0 +1,427 @@
+
+  /// Enables metric collection.
+  ///
+  /// Metric collection can only be enabled in non-release modes. It is most
+  /// useful in the profile mode where application performance is representative
+  /// of a deployed application.
+  ///
+  /// When disabled, resets collected data by calling [debugReset].
+  ///
+  /// Throws a [StateError] if invoked in release mode.
+  static set debugCollectionEnabled(bool value) {
+    if (kReleaseMode) {
+      throw _createReleaseModeNotSupportedError();
+    }
+    if (value == _collectionEnabled) {
+      return;
+    }
+    _collectionEnabled = value;
+    debugReset();
+  }
+
+  static StateError _createReleaseModeNotSupportedError() {
+    return StateError('FlutterTimeline metric collection not supported in release mode.');
+  }
+
+  static bool _collectionEnabled = false;
+
+  /// Start a synchronous operation labeled `name`.
+  ///
+  /// Optionally takes a map of `arguments`. This slice may also optionally be
+  /// associated with a [Flow] event. This operation must be finished by calling
+  /// [finishSync] before returning to the event queue.
+  ///
+  /// This is a drop-in replacement for [Timeline.startSync].
+  static void startSync(String name, { Map<String, Object?>? arguments, Flow? flow }) {
+    Timeline.startSync(name, arguments: arguments, flow: flow);
+    if (!kReleaseMode && _collectionEnabled) {
+      _buffer.startSync(name, arguments: arguments, flow: flow);
+    }
+  }
+
+  /// Finish the last synchronous operation that was started.
+  ///
+  /// This is a drop-in replacement for [Timeline.finishSync].
+  static void finishSync() {
+    Timeline.finishSync();
+    if (!kReleaseMode && _collectionEnabled) {
+      _buffer.finishSync();
+    }
+  }
+
+  /// Emit an instant event.
+  static void instantSync(String name, { Map<String, Object?>? arguments }) {
+    Timeline.instantSync(name, arguments: arguments);
+  }
+
+  /// A utility method to time a synchronous `function`. Internally calls
+  /// `function` bracketed by calls to [startSync] and [finishSync].
+  ///
+  /// This is a drop-in replacement for [Timeline.timeSync].
+  static T timeSync<T>(String name, TimelineSyncFunction<T> function,
+      { Map<String, Object?>? arguments, Flow? flow }) {
+    startSync(name, arguments: arguments, flow: flow);
+    try {
+      return function();
+    } finally {
+      finishSync();
+    }
+  }
+
+  /// The current time stamp from the clock used by the timeline in
+  /// microseconds.
+  ///
+  /// When run on the Dart VM, uses the same monotonic clock as the embedding
+  /// API's `Dart_TimelineGetMicros`.
+  ///
+  /// When run on the web, uses `window.performance.now`.
+  ///
+  /// This is a drop-in replacement for [Timeline.now].
+  static int get now => impl.performanceTimestamp.toInt();
+
+  /// Returns timings collected since [debugCollectionEnabled] was set to true, or
+  /// since the previous [debugReset], whichever was last.
```

Comment:
add: ... or the previous debugCollect ... ?

Reply:
Done.